### ISSUE TITLE
Classification Learner's `self.classes` and `fit` accuracy metric

### ISSUE THREAD

**Issue by SpaceFrostDev** [SpaceFrostDev/vysion#7]
`ClassificationLearner`'s `self.classes` currently documents the constraint that all labels in a provided dataset must be type `torch.Tensor`. However, I've implemented the use of `torchvision.datasets` and the labels provided therein are simply integers. Therefore, change   
1. The manner in which `self.classes` initializes
2. `__init__()`'s docstring for the `datasets` parameter

The accuracy metric displayed to the screen while `ClassificationLearner.fit()` is running is errorneous when training on a binary classification dataset. Currently, I assume that binary classification will be handled by the dataset by using a single value, and providing either a 0 (`False`) or 1 (`True`) as opposed to two indices, [1,0] for class 0 or [0,1] for class 1. Therefore, change:  
1. `fit()`'s accuracy metric to function with binary classification problems being represented as two indices, (i.e. the second approach)

**Comment by SpaceFrostDev** [comment]
To resolve this issue, I edited `__init__()`'s docstring for clarity and optimized the way in which `self.classes` initializes.  
 
However, upon inspecting `fit()`'s accuracy issues, I reflected back on the fundamental design assumptions of my class, and realized it would be more efficient to have simply TWO possible values for `self.task`, instead of three. All classification tasks are either SINGLE label or MULTI label, binary classification is just a subclass of single label classification. Therefore,  
1. Simplify `self.task`'s initialization to be only SINGLE or MULTI
2. Utilize `self.task`'s value to calculate error metrics on the validation set  

All changes will be implemented on branch `iss7` and all developments documented in this thread, before submitting a pull request.

**Comment by SpaceFrostDev** [comment]
Implemented both changes, with the exception that `learner.fit()`'s accuracy metric for MULTI label classification remains untested. I implemented other useful changes, like checking for label type incompatibility and prototyping the `label_map` parameter which, as of closing, has not been implemented. Single label classification is currently fully functional with it's core features.
